src/grafter/matcha/alpha.clj

```diff
@@ -3,32 +3,29 @@
   (:require [clojure.core.logic :as l :refer [fresh run*]]
             [clojure.core.logic.protocols :as lp]
             [clojure.core.logic.pldb :as pldb]
+            [clojure.spec.alpha :as s]
             [clojure.core.logic.unifier :as u]
             [clojure.string :as string]
             [clojure.walk :as walk]
-            [clojure.set :as set]))
+            [clojure.set :as set]
+            [grafter.matcha.spec :as spec]))
 
+(def ^:private ^:macro when-available #'spec/when-available)
 
 (try
   ;; avoid issue: https://github.com/Swirrl/matcha/issues/5
-
   (require '[grafter.rdf.protocols :as gp])
   (import '[grafter.rdf.protocols RDFLiteral LangString Quad])
 
-  ;; NOTE the eval here allows us to catch the compiler error we'll
-  ;; get if one of the grafter classes isn't found.
-  (eval '(extend-protocol lp/IUninitialized
-           LangString
-           (lp/-uninitialized [coll]
-             coll)
-
-           RDFLiteral
-           (lp/-uninitialized [coll]
-             coll)))
-
-  (catch clojure.lang.Compiler$CompilerException _)
   (catch java.io.FileNotFoundException _))
 
+(when-available #{LangString RDFLiteral}
+  (extend-protocol lp/IUninitialized
+    LangString
+    (lp/-uninitialized [coll] coll?)
```

Comment:
`coll?` here looks like a completion error.  It should be `coll`.

Reply:
Yup, thought that was a bit weird but didn't know where it came from